src/autorun.ts

```diff
@@ -91,72 +112,61 @@ export const computed = <T>(fn: Cb<T>): Observable<T> => new Observable(observer
         return false;
     }
 
-    const removeUnusedDeps = (ofStrength: Strength) => {
-        for (let [key, { used, subscription, strength }] of deps.entries()) {
-            if (used || strength > ofStrength) {
-                continue;
-            }
-            subscription?.unsubscribe();
-            deps.delete(key);
-        }
-    }
-
-    const runFn = () => {
-        if (hasError) {
-            return throwError(error);
-        }
+    function runFn () {
         // Mark all deps as untracked and unused
         const maybeRestoreStrength: Observable<any>[] = [];
-        for (let [key, dep] of deps.entries()) {
+        deps.forEach((dep, key) => {
             dep.track = false;
             dep.used = false;
             if (dep.strength === Strength.Normal) {
                 // Reset normal strength to weak when last run was successfull.
                 dep.strength = Strength.Weak;
                 maybeRestoreStrength.push(key);
             }
-        }
+        });
         const prevCtxt = context;
-        context = {$, _};
+        context = newCtx;
         try {
             const rsp = fn();
             removeUnusedDeps(Strength.Normal);
             return of(rsp);
         } catch (e) {
-            for (let dep of maybeRestoreStrength) {
+            maybeRestoreStrength.forEach(dep => {
                 deps.get(dep)!.strength = Strength.Normal;
-            }
+            });
             removeUnusedDeps(Strength.Weak);
-            // rethrow original errors
+
+            // suppress mid-flight interruption error
             // NOTE: check requires === if e is primitive, JS engine will try to
             //       convert ERROR_STUB to primitive
-            if (e !== ERROR_STUB) {
-                throw e;
+            if (e === HALT_ERROR) {
+                return EMPTY;
             }
-            return hasError ? throwError(error) : EMPTY;
+
+            // rethrow original errors
+            return throwError(e);
         } finally {
             context = prevCtxt;
         }
-    };
-
-    const sub = update$
-        .pipe(
-            // run fn() and completion checker instantly
-            startWith(Update.Value, Update.Completion),
-            takeWhile(u => u !== Update.Completion || anyDepRunning()),
-            switchMap(u => u === Update.Value ? runFn() : EMPTY),
-            distinctUntilChanged(),
-        )
-        .subscribe(observer);
+    }
 
-    // destroy all subscriptions
-    sub.add(() => {
-        deps.forEach((entry) => {
-            entry.subscription?.unsubscribe();
+    function removeUnusedDeps (ofStrength: Strength) {
+        deps.forEach((dep, key) => {
```

Comment:
I dropped `for-of` statements here because they generated huge & ugly code for es5 compatibility. `forEach` has native support and seem _similar_ aesthetically